src/System.Runtime.Extensions/src/System/IO/Path.WinRT.cs

```diff
@@ -11,12 +12,18 @@ namespace System.IO
 {
     public static partial class Path
     {
-        private static byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
         {
+            Debug.Assert(bytes != null);
+            Debug.Assert(byteCount >= 0);
+
             byte[] arr;
-            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteLength);
+            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteCount);
             CryptographicBuffer.CopyToByteArray(buffer, out arr);
-            return arr;
+
+            Debug.Assert(arr.Length == byteCount);
+
+            Marshal.Copy(arr, 0, new IntPtr(bytes), byteCount);
```

Comment:
@justinvp I don't think it's a good idea to add more complexity to this method (the byte array in question is only length 9, so the type cast itself could be more expensive). The only way I see to avoid these allocations is to have a different version of `GetRandomFileName` for each platform; it's unfortunate you can't do `stackalloc` for the parent function, otherwise you could write something like

```cs
// Path.cs
public static string GetRandomFileName()
{
    byte* pointer; // null on platforms where we allocate an array
    byte[] array; // null on platforms where we use stackalloc
    GetRandomCryptoBytes(9, out pointer, out array);
    if (pointer != null) DoWorkWith(pointer);
    else fixed (byte* pArray = array) DoWorkWith(pointer);
}

// Path.Windows.cs
static void GetCryptoRandomBytes(int byteCount, out byte* pointer, out byte[] unused)
{
    unused = null;
    pointer = stackalloc byte[byteCount]; // won't work, unfortunately
}

// Path.WinRT.cs
static void GetCryptoRandomBytes(int byteCount, out byte* unused, out byte[] array)
{
    unused = null;
    array = new byte[byteCount];
}
```

For now you should probably keep it the way it is.